database/src/test/scala/za/co/absa/atum/database/runs/GetPartitioningMeasuresTest.scala

```diff
@@ -0,0 +1,129 @@
+        |    "key3": "valueZ",
+        |    "key4": "valueA"
+        |  }
+        |}
+        |""".stripMargin
+    )
+
+    table("runs.partitionings").insert(
+      add("partitioning", partitioning)
+        .add("created_by", "Thomas")
+    )
+
+    val fkPartitioning: Long = table("runs.partitionings").fieldValue("partitioning", partitioning, "id_partitioning").get.get
+
+    table("runs.measure_definitions").insert(
+      add("fk_partitioning", fkPartitioning)
+        .add("created_by", "Thomas")
+        .add("measure_name", s"measure${fkPartitioning}")
+        .add("measured_columns", CustomDBType("""{"col1"}""", "TEXT[]"))
+    )
+
+    function(fncGetPartitioningMeasures)
+      .setParam("i_partitioning", partitioning)
+      .execute { queryResult =>
+        val results = queryResult.distinct.next()
+        assert(results.getInt("status").contains(11))
+        assert(results.getString("status_text").contains("OK"))
+      }
+
+    table("runs.measure_definitions").where(add("fk_partitioning", fkPartitioning)) { partitioningMeasuresResult =>
+      assert(partitioningMeasuresResult.hasNext)
+      val row = partitioningMeasuresResult.next()
+      assert(row.getString("created_by").contains("Thomas"))
+    }
+  }
+
+  test("Get partitioning measures should return error on non existing partitioning") {
+    val partitioning = JsonBString(
+      """
+        |{
+        |   "version": 1,
+        |   "keys": ["key1"],
+        |   "values": {
+        |     "key1": "value1"
+        |   }
+        |}
+        |""".stripMargin
+    )
+
+    function(fncGetPartitioningMeasures)
+      .setParam("i_partitioning", partitioning)
+      .execute { queryResult =>
+        val results = queryResult.distinct.next()
+        assert(results.getInt("status").contains(41))
+        assert(results.getString("status_text").contains("The partitioning does not exist."))
+      }
+  }
+
+  test("Get partitioning measures should return exception for partitioning with no measures") {
+    val partitioning = JsonBString(
+      """
+        |{
+        |  "keys": [
+        |    "keyA",
+        |    "keyB",
+        |    "keyC"
+        |  ],
+        |  "version": 1,
+        |  "keysToValues": {
+        |    "keyA": "valueA",
+        |    "keyB": "valueB",
+        |    "keyC": "valueC"
+        |  }
+        |}
+        |""".stripMargin
+    )
+
+    table("runs.partitionings").insert(
+      add("partitioning", partitioning)
+        .add("created_by", "Thomas")
+    )
+
+    val fkPartitioning: Long = table("runs.partitionings").fieldValue("partitioning", partitioning, "id_partitioning").get.get
+
+    function(fncGetPartitioningMeasures)
+      .setParam("i_partitioning", partitioning)
+      .execute { queryResult =>
+        val results = queryResult.distinct.next()
+        assert(results.getInt("status").contains(10))
+        assert(results.getString("status_text").contains("No measures found for the given partitioning."))
```

Comment:
you can add also these there:

`assert(!queryResult.hasNext)`

and check whether these are NULLs: `measure_name` and `measured_columns`